Add client deletion

hihuz opened the pull request:
In this PR we are adding the functionality to delete clients.

The flow is:
* on dashboard, click on a client to see details
* click on `delete client` it will open a confirmation modal
* if you click on the overlay or on cancel, nothing will happen
* if you click on `delete client` the client will be destroyed in db.

Here are some screenshots: 
<img width="751" alt="Screenshot 2019-10-25 at 11 24 35" src="https://user-images.githubusercontent.com/3253522/67561076-78d6d180-f71c-11e9-95d1-5c5600414689.png">

<img width="362" alt="Screenshot 2019-10-25 at 11 25 22" src="https://user-images.githubusercontent.com/3253522/67561080-7bd1c200-f71c-11e9-83d2-f80ba91770ba.png">

hihuz committed:
Display client details on toggle

hihuz committed:
Add actions components to client details

hihuz committed:
Improve styles for small devices

hihuz committed:
Improve spacing for client details

hihuz committed:
Use text field for secret

hihuz committed:
Merge branch 'feature/5664-console-app' into feature/5664-client-deletion

hihuz committed:
Add client deletion confirmation modal

hihuz committed:
Add mutation to delete clients

johannespfeiffer commented on console-src/src/enhancers/withOAuthClients.tsx:
```diff
@@ -90,8 +90,29 @@ class OAuthClientsProvider extends Component<Props, State> {
+    const {
+      // Currently rawQuery return type does not include client mutation results
+      // @ts-ignore
+      deleteClient: client
```
Why do we have `client` as a return type here? Was expecting it to match `Promise<void> | void`?

hihuz replied:
Hmm you're right it's not particularly useful considering we already have the client id in the same context, I have removed it here: 0082e6162f110e64014741e36e015c3617bbd432

Ideally I would also remove it from the query entirely, but the type in our schema currently do not allow it and returns a non nullable client: 
<img width="362" alt="Screenshot 2019-10-25 at 11 55 24" src="https://user-images.githubusercontent.com/3253522/67562087-5ba30280-f71e-11e9-8a88-d02866c6e3e4.png">

I will make a small PR against backend to fix that

hihuz replied:
Hmm apparently making it optional is not as simple as I thought, even if I go `@Mutation(returns => Client, { nullable: true })`, the mutation still expects me to specify some client fields as return value 🤔 

Would you be fine if we leave it like so for the time being and try to find an improvement for it later on?

johannespfeiffer replied:
Sure

hihuz committed:
Do not use client deletion return value

hihuz committed:
Do not try to close modal after client deletion

Client is removed from the list when deleted so the setState to close the modal was a noop

hihuz committed:
Use proper key for ClientList
